### PR TITLE
bpf: Move calls map to temporary location and remove after filter rep…

### PR DESCRIPTION
**Pull request by tgraf**
Removing a BPF map cause it to lose all state immediately. Current code is
removing the tail call map just before replacing the filter which could lead to
a short race window where tail calls would fail in the old program just before
it is replaced with the new program.

Fixes: 95a2c8aea ("bpf: Remove old calls map when inserting endpoint programs")

Signed-off-by: Joe Stringer <joe@covalent.io>
Signed-off-by: Thomas Graf <thomas@cilium.io>

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Review by joestringer — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
test-me-please

File: bpf/join_ep.sh [hunk 1]
```diff
@@ -58,9 +58,11 @@ fi
...
 bpf_compile bpf_lxc.c bpf_lxc.o obj
 
-# Unlink the old calls map so we can upgrade it
-CALLS_MAP=cilium_calls_$EPID
-rm "/sys/fs/bpf/tc/globals/$CALLS_MAP" 2> /dev/null || true
+CALLS_MAP="/sys/fs/bpf/tc/globals/cilium_calls_${EPID}"
 
+# Move the old calls map to a temporary location so it can be removed after the
+# program has been replaced
+mv "$CALLS_MAP" "${CALLS_MAP}.old" 2> /dev/null || true
```
Comment (aanm) [inline bpf/join_ep.sh]: If the any of the `tc` commands bellow fail, should we revert this `mv`?

Comment (tgraf) [reply bpf/join_ep.sh]: Good point

Comment (tgraf) [reply bpf/join_ep.sh]: fixed

**Review by ianvernon — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Pull request merged**